app/models/task.rb

```diff
@@ -45,20 +45,40 @@ class Task < ApplicationRecord
   scope :min_stars, ->(stars) { average_rating.where('COALESCE(avg_rating, 0) >= ?', stars) }
   scope :sort_by_average_rating_asc, -> { average_rating.order(average_rating: 'ASC') }
   scope :sort_by_average_rating_desc, -> { average_rating.order(average_rating: 'DESC') }
+  scope :access_level, ->(access_level) { where(access_level:) }
 
   serialize :meta_data, HashAsJsonbSerializer
 
+  enum access_level: {private: 0, public: 1}, _default: :private, _prefix: true
+
   def self.ransackable_scopes(_auth_object = nil)
-    %i[created_before_days min_stars]
+    %i[created_before_days min_stars access_level]
   end
 
   def self.ransackable_attributes(_auth_object = nil)
     %w[title description programming_language_id created_at]
   end
 
   def can_access(user)
+    author?(user)
```

Comment:
While I haven't had a detailed look, I at least identify two discussions in this thread:

1) The suggestion about using cancan methods from the model, without re-defining them. I would like to have that, as it would remove the duplication. However, since it is still used in the ProFormA service, this would require us touching other parts as well.
2) The question about moving the method to the `user.rb`, rather than the `task.rb`. Indeed, this is the part we talked about. At first, my impression was to move it to the user as well. However, when thinking about how to implement that (and the semantics), I wasn't sure and rather thought of making it similar to the existing method and therefore keeping it with the task.

Reply:
Concerning 2: I think I would prefer adding a `can` method to `user` and use that to implement a `is_accessible_by` in `task`